ReSharper.FSharp/src/FSharp.Psi/src/Resolve/FSharpParameterUtil.cs

```diff
@@ -90,12 +90,24 @@ public static class FSharpParameterUtil
         IMemberSignature memberSignature => GetParameterNames(((IMemberSignatureOrDeclaration)memberSignature)
           .ReturnTypeInfo.ReturnType),
 
-        IUnionCaseDeclaration ucDecl => new[] { ucDecl.Fields.Select(t => t.SourceName) },
+        IUnionCaseDeclaration ucDecl => new[] { ProcessUnionCaseFieldNames(ucDecl) },
+
         _ => EmptyList<string[]>.Enumerable
       })
       .Select(t => t.ToIReadOnlyList())
       .ToIReadOnlyList();
 
+    private static IEnumerable<string> ProcessUnionCaseFieldNames(IUnionCaseDeclaration unionsCase)
+    {
+      var fields = unionsCase.Fields;
+      var fieldsCount = fields.Count;
+
+      return fields.Select((field, i) =>
+        field.SourceName == SharedImplUtil.MISSING_DECLARATION_NAME
+          ? fieldsCount > 1 ? $"Item{i + 1}" : "Item"
+          : field.SourceName);
```

Comment:
I think there's existing code doing exactly this somewhere. Is it possible to reuse it?